meja/test.meja

```diff
@@ -0,0 +1,20 @@
+let x = 15;
+
+let y = x;
+
+let z = (x);
+
+let a = (15);
+
+let b = (x; y);
+
+let c = {x; y;};
+
+let d = fun (x) => { x; };
+
+let e = let e = let e = x; e; e;
+
+let f =
+  let f = fun (x) => { x; };
+  let g = (d f) f;
```

Comment:
you don't have to fix it in this PR, but can you make function application "reason-style" aka: `d(f)(f)` instead of `(d f) f`.

Reason just uses an extra pair of parens for representing tuples. So `a(x, y)` is the curried version and `a((x, y))` is the tupled one.

Reply:
D'oh! That actually makes some of the parsing simpler for some later things, I can't believe I missed that. I'll drop that in the next PR.